ArcGISRuntimeSDKQt_QMLSamples/Scenes/ChangeAtmosphereEffect/ChangeAtmosphereEffect.qml

```diff
@@ -0,0 +1,126 @@
+// [WriteFile Name=ChangeAtmosphereEffect, Category=Scenes]
+// [Legal]
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+import Esri.ArcGISRuntime 100.5
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    states: [
+        State {
+            when: noAtmosphereButton.checked
+            PropertyChanges {
+                target: sceneView
+                atmosphereEffect: Enums.AtmosphereEffectNone
```

Comment:
Just wondering if you can do the same state logic for the C++ sample? You would need to make the `Q_INVOKABLE` methods into properties I guess:

`Q_PROPERTY(bool atmosphereEffectNone READ atmosphereEffectNone  WRITE setAtmosphereEffectNone NOTIFY atmospeherEffectChanged)`

Reply:
Decided to upgrade with one Q_PROPERTY that takes a redeclaration of the enum type.

It is slightly uglier than I would like - but I'd argue it should be allowed in. I don't think it is unreasonable people would want to use enums across the C++/QML boundary and at least here is an example on how to do so.